src/world.rs

```diff
@@ -48,64 +48,50 @@ impl WorldMap{
         
     }
 
-    pub(crate) fn get_east(&mut self, x: isize, y: isize) -> Option<Entity>{
-        let id: Option<&usize> = self.chunk_ids.get(&(x+1, y));
-        if let Some(id) = id {
-            let entity: Option<&Entity> = self.chunks.get(id);
-            if let Some(entity) = entity {
-                Some(*entity)
-            }else{
-                None
-            }
-        }else{
-            None
-        }
+    pub(crate) fn get_east(&self, x: isize, y: isize) -> Option<Chunk>{
+        self.get_chunk(x + 1, y)
     }
     
-    pub(crate) fn get_west(&mut self, x: isize, y: isize) -> Option<Entity>{
-        let id: Option<&usize> = self.chunk_ids.get(&(x-1, y));
-        if let Some(id) = id {
-            let entity: Option<&Entity> = self.chunks.get(id);
-            if let Some(entity) = entity {
-                Some(*entity)
-            }else{
-                None
-            }
-        }else{
-            None
-        }
+    pub(crate) fn get_west(&mut self, x: isize, y: isize) -> Option<Chunk>{
+        self.get_chunk(x - 1, y)
     }
     
-    pub(crate) fn get_north(&mut self, x: isize, y: isize) -> Option<Entity>{
-        let id: Option<&usize> = self.chunk_ids.get(&(x, y+1));
-        if let Some(id) = id {
-            let entity: Option<&Entity> = self.chunks.get(id);
-            if let Some(entity) = entity {
-                Some(*entity)
-            }else{
-                None
-            }
-        }else{
-            None
-        }
+    pub(crate) fn get_north(&mut self, x: isize, y: isize) -> Option<Chunk>{
+        self.get_chunk(x, y + 1)
     }
     
-    pub(crate) fn get_south(&mut self, x: isize, y: isize) -> Option<Entity>{
-        let id: Option<&usize> = self.chunk_ids.get(&(x, y-1));
-        if let Some(id) = id {
-            let entity: Option<&Entity> = self.chunks.get(id);
-            if let Some(entity) = entity {
-                Some(*entity)
-            }else{
-                None
-            }
-        }else{
-            None
-        }
+    pub(crate) fn get_south(&mut self, x: isize, y: isize) -> Option<Chunk>{
+        self.get_chunk(x, y - 1)
     }
 
 }
     
 pub(crate) fn logical_to_rendering(x: isize, y: isize) -> (f32, f32){
     (x as f32 * WIN_W, y as f32 * WIN_H)
+}
+
+pub(crate) struct GameProgress{
```

Comment:
I like this struct, but why underscores? Just to silence warnings or? The fields are already implicitly private, and the `_` naming convention is annoying to type, so I'd refactor `_` out unless it's here for warning suppression (but even in that case, we could use a temporary `#[allow(unused)]`)

Reply:
silence warnings, and to show explicitly that this is not being used. In the spirit of commit often and commit little, function of game progress will not be introduced by this PR.